pkg/contexts/ocm/blobhandler/handlers/generic/mvn/blobhandler.go

```
@@ -0,0 +1,166 @@
+	"github.com/open-component-model/ocm/pkg/utils/tarutils"
+)
+
+const BLOB_HANDLER_NAME = "ocm/" + resourcetypes.MVN_ARTIFACT
+
+type artifactHandler struct {
+	spec *Config
+}
+
+func NewArtifactHandler(repospec *Config) cpi.BlobHandler {
+	return &artifactHandler{repospec}
+}
+
+var log = logging.DynamicLogger(identity.REALM)
+
+func (b *artifactHandler) StoreBlob(blob cpi.BlobAccess, resourceType string, hint string, _ cpi.AccessSpec, ctx cpi.StorageContext) (cpi.AccessSpec, error) {
+	// check conditions
+	if b.spec == nil {
+		return nil, nil
+	}
+	mimeType := blob.MimeType()
+	if resourcetypes.MVN_ARTIFACT != resourceType {
+		log.Debug("not a MVN artifact", "resourceType", resourceType)
+		return nil, nil
+	}
+	if mime.MIME_TGZ != mimeType {
+		log.Debug("not a tarball, can't be a complete mvn GAV", "mimeType", mimeType)
+		return nil, nil
+	}
+	if b.spec.Url == "" {
+		return nil, fmt.Errorf("MVN repository url not provided")
+	}
+
+	// setup logger
+	log := log.WithValues("repository", b.spec.Url)
+	// identify artifact
+	artifact := mvn.DeSerialize(hint)
+	log = log.WithValues("groupId", artifact.GroupId, "artifactId", artifact.ArtifactId, "version", artifact.Version)
+	log.Debug("identified")
+
+	blobReader, err := blob.Reader()
+	if err != nil {
+		return nil, err
+	}
+	defer blobReader.Close()
+	tempFs, err := tarutils.ExtractTgzToTempFs(blobReader)
+	if err != nil {
+		return nil, err
+	}
+	defer vfs.Cleanup(tempFs)
+	files, err := tarutils.ListSortedFilesInDir(tempFs, "", false)
+	if err != nil {
+		return nil, err
+	}
+	for _, file := range files {
+		log.Debug("uploading", "file", file)
+		artifact = artifact.ClassifierExtensionFrom(file)
+
+		readHash, err := tempFs.Open(file)
+		if err != nil {
+			return nil, err
+		}
+		defer readHash.Close()
+		// MD5 + SHA1 are still the most used ones in the mvn context
+		hr := iotools.NewHashReader(readHash, crypto.SHA256, crypto.SHA1, crypto.MD5)
+		_, _ = hr.CalcHashes()
```

Shouldn't ignore the error. ;)